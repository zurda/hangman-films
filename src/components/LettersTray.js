import React from 'react'
import styled from '@emotion/styled'
import { colours } from '../style/shared'
import { upprecaseLettersArr } from '../helpers'

const LettersContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: 16px auto;
`

const Char = styled.button`
  padding: 8px;
  font-weight: bold;
  margin: 2px;
  width: 10%;
  background-color: ${(props) => (props.disabled ? 'white' : colours.pictonBlue)};
  color: ${(props) => (props.disabled ? colours.diamond : 'white')};
  box-shadow: 0px 3px 15px rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  border: 0;
  @media (min-width: 420px) {
    font-size: 18px;
  }
`

const LettersTray = ({ guessedLetters, onClickHandler }) => (
  <LettersContainer>
    {upprecaseLettersArr.map((char, i) => (
      <Char
        onClick={(e) => onClickHandler(e.target.innerHTML)}
        key={i}
        value={char}
        disabled={guessedLetters.indexOf(char) > -1}
      >
        {char}
      </Char>
    ))}
  </LettersContainer>
)

export default LettersTray
