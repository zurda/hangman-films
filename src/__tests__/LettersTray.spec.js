import React from 'react'
import { create } from 'react-test-renderer'
import LettersTray from '../components/LettersTray'

const guessedLetters = [' ', 'a', 'b', 'c']

describe('LettersTray component', () => {
  test('Matches the snapshot', () => {
    const lettersTray = create(
      <LettersTray
        guessedLetters={guessedLetters}
        onClickHandler={(char = 'd') => console.log()}
      />
    )
    expect(lettersTray.toJSON()).toMatchSnapshot()
  })
})
