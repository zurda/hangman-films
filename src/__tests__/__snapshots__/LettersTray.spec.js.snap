// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LettersTray component Matches the snapshot 1`] = `
<div
  className="css-zr8jku"
>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="A"
  >
    A
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="B"
  >
    B
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="C"
  >
    C
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="D"
  >
    D
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="E"
  >
    E
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="F"
  >
    F
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="G"
  >
    G
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="H"
  >
    H
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="I"
  >
    I
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="J"
  >
    J
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="K"
  >
    K
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="L"
  >
    L
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="M"
  >
    M
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="N"
  >
    N
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="O"
  >
    O
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="P"
  >
    P
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="Q"
  >
    Q
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="R"
  >
    R
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="S"
  >
    S
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="T"
  >
    T
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="U"
  >
    U
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="V"
  >
    V
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="W"
  >
    W
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="X"
  >
    X
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="Y"
  >
    Y
  </button>
  <button
    className="css-46x0g4"
    disabled={false}
    onClick={[Function]}
    value="Z"
  >
    Z
  </button>
</div>
`;
